utils/fill_nines.h

```diff
@@ -1,8 +1,11 @@
 #ifndef UTILS_FILL_NINES_H_
 #define UTILS_FILL_NINES_H_
 
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
```

Comment:
I tested on node0 and they don't need to include explicitly. Could you check is this necessary?